rmf_site_format/src/location.rs

```diff
@@ -19,67 +19,32 @@ use crate::*;
 #[cfg(feature = "bevy")]
 use bevy::prelude::{Bundle, Component, Deref, DerefMut, Entity};
 use serde::{Deserialize, Serialize};
-
-#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
-pub enum LocationTag {
-    Charger,
-    ParkingSpot,
-    HoldingPoint,
-    SpawnRobot(Model),
-    Workcell(Model),
-}
-
-impl LocationTag {
-    pub fn label(&self) -> &'static str {
-        match self {
-            Self::Charger => "Charger",
-            Self::ParkingSpot => "Parking Spot",
-            Self::HoldingPoint => "Holding Point",
-            Self::SpawnRobot(_) => "Spawn Robot",
-            Self::Workcell(_) => "Workcell",
-        }
-    }
-
-    pub fn is_charger(&self) -> bool {
-        matches!(self, Self::Charger)
-    }
-    pub fn is_parking_spot(&self) -> bool {
-        matches!(self, Self::ParkingSpot)
-    }
-    pub fn is_holding_point(&self) -> bool {
-        matches!(self, Self::HoldingPoint)
-    }
-    pub fn spawn_robot(&self) -> Option<&Model> {
-        match self {
-            Self::SpawnRobot(model) => Some(model),
-            _ => None,
-        }
-    }
-    pub fn workcell(&self) -> Option<&Model> {
-        match self {
-            Self::Workcell(model) => Some(model),
-            _ => None,
-        }
-    }
-}
+use std::collections::{BTreeSet, HashMap, HashSet};
 
 #[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
 #[cfg_attr(feature = "bevy", derive(Bundle))]
 pub struct Location<T: RefTrait> {
     pub anchor: Point<T>,
+    #[serde(flatten)]
     pub tags: LocationTags,
     pub name: NameInSite,
     pub graphs: AssociatedGraphs<T>,
 }
 
-#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
-#[serde(transparent)]
-#[cfg_attr(feature = "bevy", derive(Component, Deref, DerefMut))]
-pub struct LocationTags(pub Vec<LocationTag>);
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Default)]
+#[cfg_attr(feature = "bevy", derive(Component))]
+pub struct LocationTags {
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub charger: Option<String>,
```

Comment:
What does `String` contain (since it was bool before)? Is it meant to be something like a `json` that will be sent to the robot, or just a human readable description? This could also be "documented" in the widget through hover text or a label. +100 for the change to a `LocationTags` struct since the `Vec<LocationTag>` Component made tag search and update a bit painful before.

Reply:
I'm still deliberating this. For now it's a placeholder meant for us to experiment with. My current expectation is it can be a `json` containing information about what type of charger it is. E.g.
```
{
  "brand": "Xnergy",
  "height_range": [0.1, 2.0],
  "current": 2.0,
  "voltage": 15
}
```
then there can be application logic somewhere to decide which robots are compatible with the charger.